src/test/OptionSettlement.t.sol

```diff
@@ -470,38 +470,65 @@ contract OptionSettlementTest is Test, NFTreceiver {
     // **********************************************************************
 
     function testEventNewOptionType() public {
-        vm.expectEmit(false, true, true, true);
+        IOptionSettlementEngine.Option memory optionInfo = IOptionSettlementEngine.Option({
+            underlyingAsset: DAI_A,
+            exerciseTimestamp: testExerciseTimestamp,
+            expiryTimestamp: testExpiryTimestamp,
+            exerciseAsset: WETH_A,
+            underlyingAmount: testUnderlyingAmount,
+            settlementSeed: 0,
+            exerciseAmount: testExerciseAmount,
+            nextClaimId: 1
+        });
+
+        bytes20 optionHash = bytes20(keccak256(abi.encode(optionInfo)));
+        uint160 optionKey = uint160(optionHash);
+        uint256 expectedOptionId = uint256(optionKey) << 96;
+
+        vm.expectEmit(true, true, true, true);
         emit NewOptionType(
-            999, WETH_A, DAI_A, testExerciseAmount, testUnderlyingAmount, testExerciseTimestamp, testExpiryTimestamp, 1
+            expectedOptionId,
+            WETH_A,
+            DAI_A,
+            testExerciseAmount,
+            testUnderlyingAmount,
+            testExerciseTimestamp,
+            testExpiryTimestamp,
+            1
             );
 
-        engine.newOptionType(
-            IOptionSettlementEngine.Option({
-                underlyingAsset: DAI_A,
-                exerciseTimestamp: testExerciseTimestamp,
-                expiryTimestamp: testExpiryTimestamp,
-                exerciseAsset: WETH_A,
-                underlyingAmount: testUnderlyingAmount,
-                settlementSeed: 0,
-                exerciseAmount: testExerciseAmount,
-                nextClaimId: 1
-            })
-        );
+        engine.newOptionType(optionInfo);
     }
 
-    function testEventWrite() public {
+    function testEventWriteWhenNewClaim() public {
         uint256 expectedFeeAccruedAmount = ((testUnderlyingAmount / 10_000) * engine.feeBps());
 
         vm.expectEmit(true, true, true, true);
         emit FeeAccrued(WETH_A, ALICE, expectedFeeAccruedAmount);
 
-        vm.expectEmit(true, true, true, false);
-        emit OptionsWritten(testOptionId, ALICE, 999, 1);
+        vm.expectEmit(true, true, true, true);
+        emit OptionsWritten(testOptionId, ALICE, 0, 1);
 
         vm.prank(ALICE);
         engine.write(testOptionId, 1);
     }
 
+    function testEventWriteWhenExistingClaim() public {
+        uint256 expectedFeeAccruedAmount = ((testUnderlyingAmount / 10_000) * engine.feeBps());
+
+        vm.prank(ALICE);
+        uint256 claimId = engine.write(testOptionId, 1);
+
+        vm.expectEmit(true, true, true, true);
+        emit FeeAccrued(WETH_A, ALICE, expectedFeeAccruedAmount);
+
+        vm.expectEmit(true, true, true, true);
+        emit OptionsWritten(testOptionId, ALICE, claimId, 1);
```

Comment:
@0xAlcibiades @neodaoist what do we think makes the most sense here to emit. Currently the index of the claim is emitted (1, 2 ,3 ,4). Do we want to emit the entire claim ID instead with the encoded option ID in the top 160b?

Reply:
(Not related to changes in this PR btw)

Reply:
Honestly I would love to talk through how dapp devs or end users might want to view and monitor these events. Need more context to have an opinion yet